### DISCUSSION TITLE
Delta to Iceberg Conversion Error - org.apache.xtable.exception.ReadException: Failed to iterate through Delta data files

### DISCUSSION THREAD

**Discussion by surajsubramani22** [apache/incubator-xtable#472] [Q&A]
Facing the below issue. Any suggestions on what I might have missed? 
![image](https://github.com/apache/incubator-xtable/assets/113149251/7a8476c0-e2a3-4eac-a35c-1e17428a464d)

Config File: 
![image](https://github.com/apache/incubator-xtable/assets/113149251/6fc7f3d7-5aa6-49c0-a376-ee23dc6b0400)

Sample Delta Table on S3 reads okay.  
![image](https://github.com/apache/incubator-xtable/assets/113149251/95dc7c7c-66dc-48ce-ba78-7a3f566b5830)

Followed the steps as detailed here -> https://xtable.apache.org/docs/how-to/#create-dataset

**Comment by the-other-tim-brown** [comment]
@surajsubramani22 it looks like the stats on the files are null within the delta commit. Is it possible the stats were not added as part of the commit?

**Reply by surajsubramani22** [reply]
Thanks @the-other-tim-brown , Are you referring to this? I’m able to see the below contents in the file and able to read the delta table fine. 
![image](https://github.com/apache/incubator-xtable/assets/113149251/7423afa5-5624-4a62-8aa0-471f3ee39914)

**Reply by the-other-tim-brown** [reply]
Yes, that is what I was referring to. I think we need a fix in the code to account for this case. Are you interested in submitting a patch?

**Reply by surajsubramani22** [reply]
Makes sense. Tested with a delta stats file and it worked fine. With regards to a fix for the bug, It will probably take me a while to get to it unfortunately.

**Reply by the-other-tim-brown** [reply]
I have created this issue for tracking and will put up a fix shortly https://github.com/apache/incubator-xtable/issues/481

**Reply by the-other-tim-brown** [reply]
Here is the PR: https://github.com/apache/incubator-xtable/pull/482
